### PR TITLE
WP.com block editor: Deprecate CoBlocks buttons block

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

Overrides the CoBlocks Buttons block settings to make it non-insertable (so it won't collide with Core's new Buttons block).

Further reading: paYKcK-rn-p2

#### Testing instructions

* Apply D37423-code, sandbox `widgets.wp.com` and a simple site.
* Load the block editor.
* Try finding the CoBlocks 'Buttons' block in the block inserter: It shouldn't be there.
* Previously created posts that include the CoBlocks 'Buttons' block should continue to work.

### CODE REVIEW

File: apps/wpcom-block-editor/src/default/features/rich-text.js [hunk 1]
```diff
@@ -75,9 +76,11 @@ const ConnectedRichTextJustifyButton = compose(
 	ifCondition( props => 'core/paragraph' === props.blockName )
 )( RichTextJustifyButton );
 
-registerFormatType( 'wpcom/justify', {
-	title: wpcomGutenberg.richTextToolbar.justify,
-	tagName: 'p',
-	className: null,
-	edit: ConnectedRichTextJustifyButton,
+domReady( function() {
+	registerFormatType( 'wpcom/justify', {
```
Comment (Addison-Stavlo) [inline apps/wpcom-block-editor/src/default/features/rich-text.js]: This is to ensure that this registration runs after the Filter has changed the `inserter` property for the coblocks/buttons?

I was initially wondering if waiting until `domReady` to register may have any adverse side effects, but I'm not really noticing any in testing so it seems to be working well.

**Review by Addison-Stavlo — APPROVED** [review]
These changes look good and seem to work well.

I tested this on my sandbox.  It does remove the option to insert the coblocks/buttons, but it does still allow me to use and update any coblocks/buttons that I had inserted previous to the change.

The p2 consensus seemed to be to go with the deprecation approach, and this seems to do that well!

File: apps/wpcom-block-editor/src/default/features/rich-text.js (hunk 1 above)
Comment (ockham) [reply apps/wpcom-block-editor/src/default/features/rich-text.js]: D37423-code has more context:

> If order for the filter to run at the correct time, we also need to enqueue the script in the header rather than the footer, so CoBlocks are registered afterwards.

>  [...] just realized that the justify format is missing when this patch is applied, which is likely caused by enqueuing the scripts on the head rather than on the footer. I'll investigate if we can register the justify format in a way that can be loaded before the editor scripts but executed afterwards.

> `domReady` did the trick: https://github.com/Automattic/wp-calypso/pull/38724/commits/4c3647a7a70fbaf356adea844578f2b95f49ca88

**Comment by mmtr** [comment]
We might want to try using [`addBlockTypes`](https://github.com/WordPress/gutenberg/blob/8fcbed7c5b431dfa1127af7f914bb31b5479d5de/packages/blocks/src/store/actions.js#L13) for overriding the CoBlocks Buttons block settings instead of using the `blocks.registerBlockTypes` filter. Despite its name, I think `addBlockTypes` override the block types if we pass an existing block name.

That way, we don't need to change the way the scripts are enqueued (they can remain in the footer) and avoid the concerns raised in https://github.com/Automattic/jetpack/pull/14310#issuecomment-572250170

**Comment by mmtr** [comment]
> We might want to try using [`addBlockTypes`](https://github.com/WordPress/gutenberg/blob/8fcbed7c5b431dfa1127af7f914bb31b5479d5de/packages/blocks/src/store/actions.js#L13) for overriding the CoBlocks Buttons block settings instead of using the `blocks.registerBlockTypes` filter

I finally re-registered the block using `unregisterBlockType` and `registerBlockType`, so the script can be now loaded in the footer.

**Comment by Addison-Stavlo** [comment]
Nice update!

So instead of delaying all block registration until `domReady`, now we are back to registering blocks as normal and just re-registering that one specific block on `domReady` ?  That seems to be more reasonable!

**Review by Addison-Stavlo — APPROVED** [review]
Nice update.  I just re-tested on the sandbox and things behave well as before.  Prevents inserting the deprecated block but I am still able to use and update one that was added before the deprecation.

**Comment by kwight** [comment]
LGTM; verified CoBlocks buttons are not available by inserters, and existing posts with CoBlocks buttons load properly.

Screen Shot 2020-01-09 at 1 48 16 PM

**Review by kwight — APPROVED** [review]

File: apps/wpcom-block-editor/src/wpcom/features/deprecate-coblocks-buttons.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable import/no-extraneous-dependencies */
+
+/**
+ * External dependencies
+ */
+import { getBlockType, registerBlockType, unregisterBlockType } from '@wordpress/blocks';
+import domReady from '@wordpress/dom-ready';
```
Comment (sirreal) [inline apps/wpcom-block-editor/src/wpcom/features/deprecate-coblocks-buttons.js]: I looked at a related Jetpack PR and noticed the `domReady` script dependency was added. Is that related to this change? Are these dependencies externals? If so, do we have a way of ensuring that the dependencies are properly enqueued when the script is served from the widgets URL?

Comment (simison) [reply apps/wpcom-block-editor/src/wpcom/features/deprecate-coblocks-buttons.js]: Good question. Jetpack always loads the latest bundle from widgets URL, no matter what version of Jetpack we're running.

Comment (mmtr) [reply apps/wpcom-block-editor/src/wpcom/features/deprecate-coblocks-buttons.js]: > I looked at a related Jetpack PR and noticed the `domReady` script dependency was added. Is that related to this change? 

Do you mean https://github.com/Automattic/jetpack/pull/14310? If so, yeah, it was related to a previous commit of this PR. We changed the approach and that PR was no longer needed, since this `domReady` is used in the `wpcom` bundled script which [Jetpack already enqueues with `wp-dom-ready` as dependency](https://github.com/Automattic/jetpack/blob/572dd6488baf30ef15853ca0cbb30a8f98294f16/modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php#L306).

> Are these dependencies externals? 

Yup, they are [externalized when generating the build with Webpack](https://github.com/Automattic/wp-calypso/blob/b08d9f6422751c7f4a7e78aba0e5a9a75aa2c9b7/apps/wpcom-block-editor/webpack.config.js#L63).

> If so, do we have a way of ensuring that the dependencies are properly enqueued when the script is served from the widgets URL?

We manually maintain them in [Jetpack](https://github.com/Automattic/jetpack/blob/572dd6488baf30ef15853ca0cbb30a8f98294f16/modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php#L258-L344) and WP.com (handled in the `gutenberg_wpcom_enqueue_scripts` function of the `wp-content/plugins/gutenberg-wpcom/index.php` file).
